test_quality_of_service/test/test_deadline.cpp

```diff
@@ -58,29 +60,31 @@ TEST_F(QosRclcppTestFixture, test_deadline) {
   subscriber = std::make_shared<QosTestSubscriber>(
     "subscriber", topic, qos_profile);
 
-  // setup publishing options and callback
-  publisher->options().event_callbacks.deadline_callback =
-    [this, &last_pub_count, &total_number_of_publisher_deadline_events](
-    rclcpp::QOSDeadlineOfferedInfo & event) -> void
-    {
-      RCLCPP_INFO(publisher->get_logger(), "QOSDeadlineOfferedInfo callback");
-      total_number_of_publisher_deadline_events++;
-      // assert the correct value on a change
-      ASSERT_EQ(1, event.total_count_change);
-      last_pub_count = event.total_count;
-    };
-
-  // setup subscription options and callback
-  subscriber->options().event_callbacks.deadline_callback =
-    [this, &last_sub_count, &total_number_of_subscriber_deadline_events](
-    rclcpp::QOSDeadlineRequestedInfo & event) -> void
-    {
-      RCLCPP_INFO(subscriber->get_logger(), "QOSDeadlineRequestedInfo callback");
-      total_number_of_subscriber_deadline_events++;
-      // assert the correct value on a change
-      ASSERT_EQ(1, event.total_count_change);
-      last_sub_count = event.total_count;
-    };
+  if (rmw_implementation_str != "rmw_zenoh_cpp") {
```

Comment:
similar question from another PR. what if user application register the event callbacks with rmw_zenoh? i think that something we cannot control and application does that already. if that breaks the application with just switching into rmw_zenoh, that would be problem i think...